astroquery/heasarc/core.py

```diff
@@ -1,92 +1,246 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-from typing import Union
+import os
+
+import shutil
+import requests
+import tarfile
 import warnings
-from io import StringIO, BytesIO
+import numpy as np
 from astropy.table import Table
-from astropy.io import fits
 from astropy import coordinates
 from astropy import units as u
+from astropy.utils.decorators import deprecated, deprecated_renamed_argument
+
+import pyvo
+
+from astroquery import log
 from ..query import BaseQuery
-from ..utils import commons
-from ..utils import async_to_sync
+from ..utils import commons, async_to_sync, parse_coordinates
 from ..exceptions import InvalidQueryError, NoResultsWarning
 from . import conf
 
-__all__ = ['Heasarc', 'HeasarcClass']
-
-
-def Table_read(*args, **kwargs):
-    if commons.ASTROPY_LT_5_1:
-        return Table.read(*args, **kwargs)
-    else:
-        return Table.read(*args, **kwargs, unit_parse_strict='silent')
-
 
 @async_to_sync
 class HeasarcClass(BaseQuery):
```

Comment:
Do inherit from the VO baseclass, too, using the order in the other modules (so e.g. the session is properly get passed back to pyvo, and a couple of other small things)

Reply:
done